lib/elixir/test/elixir/calendar/date_test.exs

```diff
@@ -103,4 +105,14 @@ defmodule DateTest do
     assert Date.diff(date1, date2) == -13
     assert Date.diff(date2, date1) == 13
   end
+
+  test "sigil_D for a calendar other than Calendar.ISO" do
+    assert %Date{year: 10001, month: 1, day: 1, calendar: Calendar.Holocene} ==
+             ~D[10001-01-01 Calendar.Holocene]
```

Comment:
nitpick: I believe the convention is to either do: `assert expected == actual` or `assert actual = expected`, does that make sense? In this case it would be:

```elixir
assert ~D[10001-01-01 Calendar.Holocene] ==
         %Date{year: 10001, month: 1, day: 1, calendar: Calendar.Holocene}
```

probably not a strict rule either, just something I tend to see elsewhere.